src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/AbbreviationAsWordInNameCheck.java

```diff
@@ -583,8 +605,17 @@ private String getDisallowedAbbreviation(String str) {
             else if (abbrStarted) {
                 abbrStarted = false;
 
-                final int endIndex = index - 1;
-                result = getAbbreviationIfIllegal(str, beginIndex, endIndex);
+                final int endIndex;
+                final int allowedLength;
+                if (symbol == '_') {
+                    endIndex = index;
+                    allowedLength = allowedAbbreviationLength + 1;
```

Comment:
I am not fully following. Why do we need to pretend the property `allowedAbbreviationLength` is larger than it is? Why isn't the indexes enough?

Reply:
Cause we are passing the index of underscore, not last character that's why.

Reply:
@rnveach, you can check previous commit and it's report for better understanding